### PR TITLE
[BACKLOG-38917] Add Radar chart

### PR DESCRIPTION
**Pull request by AAravind27**
please review for https://hv-eng.atlassian.net/browse/BACKLOG-38917

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 8m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **73**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1753/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js [hunk 1]
```diff
@@ -85,6 +84,12 @@ define([
         data: categories,
         type: "scroll",
         align: "auto",
+        icon: "circle",
+        orient: "vertical",
+        right: "0",
+        top: "30",
+        itemWidth: 8,
+        itemHeight: 8,
```
[the remark this answers is not in the document]
Comment (AAravind27) [reply impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js]: Moved this change to another upcoming PR

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 6s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5240**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1753/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: impl/client/src/main/javascript/web/pentaho/data/_AbstractTable.js [hunk 2]
```diff
@@ -120,6 +120,58 @@ define([
...
+      if(colCount === 0) {
+        return "";
+      }
+
+      if(colCount === 1) {
+        return this.getFormattedValue(rowIndex, colIndexes[0]);
+      }
+
+      if(labelSep == null) {
+        labelSep = " ~ ";
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/pentaho/data/_AbstractTable.js:157]: Can be done on a later PR. It seems that labels are also joined with "~" (no spaces).

File: impl/client/src/main/javascript/web/pentaho/visual/models/Radar.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      props: [
+        {
+          name: "rows",
+          modes: [{dataType: "list"}],
+          fields: {isRequired: true}
+        },
+        {
+          name: "measures",
+          base: "pentaho/visual/role/Property",
+          modes: {dataType: ["number"]},
```
[the remark this answers is not in the document]
Comment (dcleao) [reply impl/client/src/main/javascript/web/pentaho/visual/models/Radar.js:48]: Can be done on another PR. Still not addressed.

File: impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+      var indicators = [];
+
+      // One indicator/axis per category value x measure name
+
+      visualData.categories.forEach(function(category) {
+        var categoryTableView = this._getEntityTableView(category, dataTable);
+
+        // When  category is not mapped, do not include the category label.
+        // Example: "Cars ~ "
+        var categoryLabelPrefix = category.label ? (category.label + " ~ ") : "";
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js:128]: Can be addressed on a follow-up PR. Should have a constant for this on the base class and be using it here as well as passing it to the various calls to `DataTable#getCompositeFormattedValue`.

File: impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+
+        return rowIndexByCategory;
+      }
+    },
+
+    _configureData: function(options, radarData) {
+      options.series.forEach(function(row) {
+        row.data = radarData.records;
+      });
+    },
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js:223]: Can be addressed on a follow-up PR.
This is still a bit "ugly", revealing that the base class may need to be changed to better accommodate for the various cases.

**Review by dcleao — APPROVED** [review]

File: impl/client/src/main/javascript/web/pentaho/data/_AbstractTable.js (hunk 2 above)
Comment (AAravind27) [reply impl/client/src/main/javascript/web/pentaho/data/_AbstractTable.js:157]: In webdetails, "~" with spaces is `groupedLabelSep` and without spaces is `dataSeparator`. But, in CCC's [config ](https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/pentaho/visual/config.js#L195)for Abstract class, 'groupedLabelSep' is without spaces.
image 
image

Comment (AAravind27) [reply impl/client/src/main/javascript/web/pentaho/data/_AbstractTable.js:157]: So, should I use with spaces or without spaces?

Comment (dcleao) [reply impl/client/src/main/javascript/web/pentaho/data/_AbstractTable.js:157]: CCC-based views are overriding CCC's default.
We should use what current stock views are using, which is without spaces.

File: impl/client/src/main/javascript/web/pentaho/visual/models/Radar.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (AAravind27) [reply impl/client/src/main/javascript/web/pentaho/visual/models/Radar.js:48]: Addressed https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1760/files#diff-60e081279c5f2b932dccf960d569aff186d04e7a30881826b51aab3d9e8334c2R49

File: impl/client/src/main/javascript/web/pentaho/data/_AbstractTable.js (hunk 2 above)
Comment (AAravind27) [reply impl/client/src/main/javascript/web/pentaho/data/_AbstractTable.js:157]: Addressed https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1760/files#diff-60849b9c6b08ea073c3ce0b7252d9525d43e1aedb9befcad1cfe6b8d81d9036cR157

File: impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js (hunk 4 above)
Comment (AAravind27) [reply impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js:128]: Addressed https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1760/files#diff-758fe27164ce4a5a8b841d0c48dc1052fb689ac73c22fff589a9ca9ba5151ec1R27

File: impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js (hunk 5 above)
Comment (AAravind27) [reply impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js:223]: Addressed. Data ( or in this case `radarData` ) is passed as an argument to `_buildSeries` https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1760/files#diff-5ce95ff23a939baa8f5f97a800275857cc295436cc15f09262d5fa204829528aR290 .Did the same for Funnel and Gauge.

**Pull request merged**